go/engine/login_provision.go

```diff
@@ -747,6 +776,45 @@ func (e *loginProvision) chooseDevice(m libkb.MetaContext, pgp bool) (err error)
 	}
 }
 
+func (e *loginProvision) preloadedPaperKey(m libkb.MetaContext, devices []*libkb.Device, paperKey string) error {
+	// User has requested non-interactive provisioning - first parse their key
+	keys, prefix, err := getPaperKeyFromString(m, e.arg.PaperKey)
+	if err != nil {
+		return err
+	}
+
+	// ... then match it to the paper keys that can be used with this account
+	for _, d := range devices {
+		if d.Type != libkb.DeviceTypePaper {
+			continue
+		}
+		if prefix != *d.Description {
+			continue
+		}
+
+		// use the KID to find the uid, deviceID and deviceName
```

Comment:
can only do this once, no need to do it every time through the loop, since it doesn't vary with `d`

Reply:
It only occurs once, when the key is matched.

Reply:
can you then refactor this function then to make the flow clearer?  So a little loop to find the key.  Then return `NoPaperKeysError{}` if not found. Then the logic that's currently inside the loop